File: app/src/main/java/ua/ck/zabochen/englishverbs/dagger/AppModule.java
package ua.ck.zabochen.englishverbs.dagger;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ua.ck.zabochen.englishverbs.helper.database.RealmHelper;
import ua.ck.zabochen.englishverbs.helper.notification.NotificationHelper;
import ua.ck.zabochen.englishverbs.helper.speech.SpeechHelper;

@Module
public class AppModule {

    private Context mContext;

    public AppModule(Context mContext) {
        this.mContext = mContext;
    }

    @Provides
    @Singleton
    public Context provideContext() {
        return mContext;
    }

    // Realm
    @Provides
    @Singleton
    public RealmHelper provideRealmHelper(Context context) {
        return new RealmHelper(context);
    }

    // TextToSpeech
    @Provides
    @Singleton
    public SpeechHelper provideSpeechHelper(Context context) {
        return new SpeechHelper(context);
    }

    // Notification
    @Provides
    @Singleton
    public NotificationHelper provideNotificationHelper() {
        return new NotificationHelper();
    }

}

File: app/src/main/java/ua/ck/zabochen/englishverbs/koin/AppModule.kt
package ua.ck.zabochen.englishverbs.koin

import org.koin.dsl.module.module
import ua.ck.zabochen.englishverbs.helper.database.DatabaseHelper
import ua.ck.zabochen.englishverbs.helper.notification.NotificationHelper
import ua.ck.zabochen.englishverbs.helper.speech.SpeechHelper

val appModule = module {
    single { DatabaseHelper(get()) }
    single { NotificationHelper() }
    single { SpeechHelper(get()) }
}